osu.Game.Tests/Visual/Gameplay/TestSceneSkinEditorComponentsList.cs

```diff
@@ -30,6 +39,46 @@ public void TestToggleEditor()
             }));
         }
 
+        [Test]
+        public void TestInvisibleComponent()
+        {
+            // Lift TestRuleset and ToolboxInvisibleComponent into an isolated assembly containing both types at a top level.
+            AssemblyBuilder asmBuilder = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName(Guid.NewGuid().ToString()), AssemblyBuilderAccess.Run);
+            ModuleBuilder moduleBuilder = asmBuilder.DefineDynamicModule(asmBuilder.GetName().Name!);
+
+            moduleBuilder.DefineType($"{nameof(ToolboxInvisibleComponent)}", TypeAttributes.Public | TypeAttributes.Class, typeof(ToolboxInvisibleComponent)).CreateType();
+            TypeBuilder rulesetType = moduleBuilder.DefineType("MockRuleset", TypeAttributes.Public | TypeAttributes.Class, typeof(TestRuleset));
+            Ruleset ruleset = (Ruleset)Activator.CreateInstance(rulesetType.CreateType())!;
```

Comment:
All of these skin editor tests are not very flexible for anything other than integration testing. The intention here was to create an isolated context with the expectation being that this was to be used in more tests from the start, however that didn't end up happening.

However, this is still required specifically because of the "containing both types at a top level" part. See the following code:

https://github.com/ppy/osu/blob/94cbe1838fff6ce48d01fccea2ba14ac9c5b2067/osu.Game/Skinning/SerialisedDrawableInfo.cs#L120-L128

`IsPublic == false` for nested types, such as `ToolboxInvisibleComponent`. The alternative is to change this to also check for `IsPublicNested`, but I wasn't sure whether we want that.